chemprop/train/molecule_fingerprint.py

```diff
@@ -0,0 +1,163 @@
+    args: Union[PredictArgs, TrainArgs]
+
+    # Same number of molecules must be used in training as in making predictions
+    assert train_args.number_of_molecules == args.number_of_molecules, (
+        'A different number of molecules was used in training '
+        f'model than is specified for prediction, {train_args.number_of_molecules} '
+        'smiles fields must be provided')
+
+    print('Loading data')
+    if smiles is not None:
+        full_data = get_data_from_smiles(
+            smiles=smiles,
+            skip_invalid_smiles=False,
+            features_generator=args.features_generator
+        )
+    else:
+        full_data = get_data(path=args.test_path, target_columns=[], ignore_columns=[], skip_invalid_smiles=False,
+                             args=args, store_row=True)
+
+    print('Validating SMILES')
+    full_to_valid_indices = {}
+    valid_index = 0
+    for full_index in range(len(full_data)):
+        if all(mol is not None for mol in full_data[full_index].mol):
+            full_to_valid_indices[full_index] = valid_index
+            valid_index += 1
+
+    test_data = MoleculeDataset([full_data[i] for i in sorted(full_to_valid_indices.keys())])
+
+    # Edge case if empty list of smiles is provided
+    if len(test_data) == 0:
+        return [None] * len(full_data)
+
+    print(f'Test size = {len(test_data):,}')
+
+    # Create data loader
+    test_data_loader = MoleculeDataLoader(
+        dataset=test_data,
+        batch_size=args.batch_size,
+        num_workers=args.num_workers
+    )
+
+    print(f'Encoding smiles into a fingerprint vector from a single model')
+    assert len(args.checkpoint_paths) == 1, "Fingerprint generation only supports one model, cannot use an ensemble"
+    # Load model
+    model = load_checkpoint(args.checkpoint_paths[0], device=args.device)
+
+    model_preds = model_fingerprint(
+        model=model,
+        data_loader=test_data_loader
+    )
+
+    # Save predictions
+    print(f'Saving predictions to {args.preds_path}')
+    assert len(test_data) == len(model_preds)
+    makedirs(args.preds_path, isfile=True)
+
+    # Copy predictions over to full_data
+    for full_index, datapoint in enumerate(full_data):
+        valid_index = full_to_valid_indices.get(full_index, None)
+        preds = model_preds[valid_index] if valid_index is not None else ['Invalid SMILES'] * (args.hidden_size*args.number_of_molecules)
+
+        fingerprint_columns=[f'fp_{i}' for i in range(args.hidden_size*args.number_of_molecules)]
+        for i in range(len(fingerprint_columns)):
+            datapoint.row[fingerprint_columns[i]] = preds[i]
+
+    # Write predictions
+    with open(args.preds_path, 'w') as f:
+        # writer = csv.DictWriter(f, fieldnames=full_data[0].row.keys())
+        writer = csv.DictWriter(f, fieldnames=args.smiles_columns+fingerprint_columns,extrasaction='ignore')
+        writer.writeheader()
+        # writer = csv.writer(f)
```

Comment:
We should delete all the comment lines here

Reply:
Whoops cleaning that up now